### PR TITLE
Add sqpv log field for server protocol

### PR DESCRIPTION
**Pull request by shinrich**
Preparing for H2 to origin (PR #7622).

### CODE REVIEW

File: proxy/logging/LogAccess.cc [hunk 1]
```diff
@@ -1782,6 +1782,40 @@ LogAccess::marshal_client_req_protocol_version(char *buf)
...
+  -------------------------------------------------------------------------*/
+
+int
+LogAccess::marshal_server_req_protocol_version(char *buf)
+{
+  const char *protocol_str = m_http_sm->server_protocol;
+  int len                  = LogAccess::strlen(protocol_str);
+
+  // Set major & minor versions when protocol_str is not "http/2".
+  if (::strlen(protocol_str) == 4 && strncmp("http", protocol_str, 4) == 0) {
```
Comment (SolidWallOfCode) [inline proxy/logging/LogAccess.cc:1795]: You could do
```
if ("http"_sv == protocol_str)
```
which would do the same - the comparison checks the length first.

**Comment by shinrich** [comment]
[approve ci clang-format]

**Comment by shinrich** [comment]
[approve ci]

File: proxy/logging/LogAccess.cc [hunk 2]
```diff
@@ -1782,6 +1782,40 @@ LogAccess::marshal_client_req_protocol_version(char *buf)
...
+      HTTPVersion versionObject = m_proxy_request->version_get();
+      int64_t major             = HTTP_MAJOR(versionObject.m_version);
+      int64_t minor             = HTTP_MINOR(versionObject.m_version);
+      if (major == 1 && minor == 1) {
+        protocol_str = "http/1.1";
+      } else if (major == 1 && minor == 0) {
+        protocol_str = "http/1.0";
+      } // else invalid http version
+    } else {
+      protocol_str = "*";
```
Comment (randall) [inline proxy/logging/LogAccess.cc:1806]: why `*` ? I'm not trying to be nit-picky, just not sure I've seen a field with that set before.

Comment (shinrich) [reply proxy/logging/LogAccess.cc:1806]: Ages ago when we used %<pitag> as a cheap way to get some information about whether H2 was triggering on the client side, '*' was the default value.

But you are correct, dash would seem to be more appropriate.  I'll update.

Comment (shinrich) [reply proxy/logging/LogAccess.cc:1806]: Actually marshal_client_req_protocol_version also uses '*' as the default. I can change them both.  Or do that in a later PR.

**Review by randall — APPROVED** [review]
I think it's fine to keep as `*` and change them to `-` in another PR (as that may be considered backwards incompatible)

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**